### PR TITLE
Fix missing individual owner home address error highlight

### PR DESCRIPTION
**Pull request by jasonaowen**
Short version: when the individual owner home address form is not filled in, put the error highlight on the first address line field rather than the second line.

Long version:
There was a Drools validation rule that required address line 2 to be present if either line 1 or 2 were present. That's not how we usually think of addresses working, and that mismatch was previously being "fixed" (or perhaps, more accurately, worked around) by testing (in Java code) if the first line was empty, and if so switch lines 1 and 2. This pattern was repeated in every place along the boundry between JSPs and controllers that dealt with addresses.

One of the places, dealing with the home residence address of the individual ownership interest, did not properly switch the two address lines, leading to the second address line being marked as in error when neither are filled, rather than the first address line.

I found the logic around switching the address lines very confusing, and so rather than fix the symptom, fix the underlying problem: update the Drools rule to prioritize address line one rather than address line two, and remove all the complexity around switching address lines everywhere else.

To do that, I did a bunch of refactoring. First, I [cleaned up the whitespace in the rules file](https://github.com/OpenTechStrategies/psm/commit/dc1285d706e5d43fd51298fbd23085c62eb6b3fc?w=1); next, I extracted a bunch of helper methods, as most of the code that dealt with addresses was copy-pasted for each form; then, I fixed the rule and deleted the code working around the previously broken rule.

---

To test this, I went through all the places I could think of that dealt with and validated addresses:
- Individual provider private practice screenshot 2017-09-26 16 42 20
- Individual provider primary practice 
screenshot 2017-09-26 15 53 53
- Individual provider additional practice 
screenshot 2017-09-26 16 07 38
- Organization provider practice screenshot 2017-09-26 16 10 06
- Individual person ownership or control interest - home residence address 
screenshot 2017-09-26 16 12 14
- Business ownership or control interest- business address 
screenshot 2017-09-26 16 13 00
- Individual personal care assistant residential address 
screenshot 2017-09-26 16 16 10 - personal care assistant

---

Issue #8 Reduce code duplication
Issue #292 Label all required fields

### CODE REVIEW

File: psm-app/services/src/main/java/gov/medicaid/binders/BaseFormBinder.java [hunk 1]
```diff
@@ -525,4 +525,25 @@ protected AddressType readPrimaryAddress(HttpServletRequest request) {
         address.setCounty(param(request, "county"));
         return address;
     }
+
+    AddressType readPrefixedAddress(
+            HttpServletRequest request,
+            String prefix
+    ) {
```
[the remark this answers is not in the document]
Comment (jasonaowen) [reply psm-app/services/src/main/java/gov/medicaid/binders/BaseFormBinder.java]: The various form binders help move data in both directions.

When a page is rendered, existing data is attached to the `ModelView` with the overloaded `attr` functions, which the JSPs read and use to fill in the page (either `<span>`s and the like, or as already-filled-in `<input>` fields).

When the user clicks "Next", the form data is sent back in a POST request, and the form binders pull the data back out with `param(request, 'name')` calls.

So, to answer your question: we pass a `request` when we're getting data from the user; we pass a `mv` when we're sending data to the user.
